### DISCUSSION TITLE
Nagios XI 2024R1.3.1 is compatible with NSCLIENT 0.3.9.328?

### DISCUSSION THREAD

**Discussion by arturo-abbido** [mickem/nscp#841] [General]
We have Nagios XI 5.9.1 and all windows clients are using NSCLIENT 0.3.9.328.
We are planning to upgrade our Nagios to 2024R1.3.1 but wondering what's gonna happen with our client machines.
This client version also uses nsc.ini file unlike the NSCLIENT 0.5.3004 uses nsclient.ini

Because of these difference, we worried that will stops all of the monitoring.
Sadly, we dont have dev or test environment. We have 200+ client machines

I asked this question to Nagios but they cant answer the question since this is a third part app

**Comment by mickem** [comment]
There was (is?) an automatic upgrade that handles most common cases so most likely if you upgrade from 0.3.x to 0.4.x and then 0.5.x and 0.6.x it will "likely just work".
Also consider the cost of failure versus the effort involved in upgrading step-by-step.

Liley the machines have the same config and that config is simple. So a fresh install with a new config might be a simpler and accept some "down time" of the monitoring but it depends if it is a mission critical system or not (i.e. the cost of failure).
With newer version of NSClient++ you can apply configuration during install which makes things simpler.

[There is a guide to upgrading from 0.3.0 to 0.4.x here.](https://nsclient.org/docs/howto/03x_migration/)

Beware that if you are using check_nt you will likely want to upgrade to check_nrpe as that is a more secure protocol.

Also if you provide more details (i.e. the old config file) I could investigate what will happen when you upgrade.
And if you want, I could help you out on a consulting bases to do the upgrade. If soe feel free to contact me at michael@medin.name

**Comment by egalstad** [comment]
Hi Michel and Arduro - I'll ask our techs to do a test to upgrading to the latest version of NSClient++ and do a writeup on the process and any issues they (hopefully don't run into).  Best regards!  Ethan

**Comment by egalstad** [comment]
FYI Michael, I just sent you an email - hope all is well!

**Comment by mickem** [comment]
Hello,

Just thought I'd follow up that there is new version here: Release Preview for 0.6.1.0 · mickem/nscp <https://github.com/mickem/nscp/releases/tag/0.6.1.0>
Which supports version 4 of the NRPE protocol as per: NRPE 4 support - NSClient++ <https://nsclient.org/news/2024-01-04/>

I expect this version to be "release as non-pre-release" in the next 30 days.
The reason it is a prerelease is that some rarely used features are not available (most notably the web ui for configuration)

// Michael Medin

Fre den 29 nov 2024 kl 02:57 skrev Ethan Galstad:

**Comment by aaronagios** [comment]
Hi @arturo-abbido,

If the Nagios team could verify that Nagios XI 2024R1.3.1 can work with NSCLIENT 0.3.9.328 would that be sufficient for you?

Kind Regards,

Aaron

**Comment by mickem** [comment]
It should work fine, I have verified that check_nrpe can work both with and without certificates with latest version of NSClient++
